core-libs/setup/ssr/optimized-engine/optimized-ssr-engine.ts

```diff
@@ -80,14 +90,26 @@ export class OptimizedSsrEngine {
       if (typeof value === 'function') {
         return value.toString();
       }
+      if (this.isServerLogger(value)) {
+        return value.constructor.name;
+      }
       return value;
     };
 
     const stringifiedOptions = JSON.stringify(this.ssrOptions, replacer, 2);
-    this.log(
-      `[spartacus] SSR optimization engine initialized with the following options: ${stringifiedOptions}`,
-      false
-    );
+    // This check has been introduced to avoid breaking changes. Remove it in Spartacus version 7.0
+    const message = this.ssrOptions.logger
+      ? `[spartacus] SSR optimization engine initialized`
+      : `[spartacus] SSR optimization engine initialized with the following options: ${stringifiedOptions}`;
+    this.log(message, true, {
+      options: {
+        ...this.ssrOptions,
+        logger:
+          this.ssrOptions.logger === true
+            ? this.ssrOptions.logger
+            : this.ssrOptions.logger?.constructor.name,
```

Comment:
[MEDIUM]

Let's make it fully safe against unexpected values

```suggestion
            : this.ssrOptions.logger?.constructor?.name,
```